### ISSUE TITLE
Demo Scenario for Compose

### ISSUE THREAD

**Issue by nichjones1** [containers/podman-desktop#3345]
### Goals:
Blog post + Demo materials on https://github.com/redhat-developer/podman-desktop-demo

### Goals of the demo:
- Podman and Podman Desktop does not break existing developer flows which rely on Compose
- It is easy to configure
- Re-emphasis that we are providing a nice UI for working with compose

### Sample code
https://github.com/redis-developer/beats-by-redis

### Features we want to show
Being able to install docker compose system wide / from a fresh install without having to go find the binary?
Standard features: Start/Stop/Restart/Delete/Logs/Inspect/Terminal/Summary

### High Level Flow
**Initial steps:**
- Starting from an existing application with an “docker compose up” from the IDE
- → Show that it works, that in the terminal I can see the image built and started
- Jump into Podman Destkop
- → Show the UI with the list of containers, I can see my group of compose containers
- → I can inspect, we do a tour of the features, logs, ssh, start stop delete

**Second steps:**
- What if I want to go to kubernetes, because that’s the platform I want to deploy my app onto
- → Show the kubernetes YAML tabs, it helps getting started
- And then, what if you were using kubernetes yamls the way you use compose
- → we switch the source code from compose to yamls (that means, we prepared the yamls before)
- → we explain the yamls, how they are defined, how they link to dockerfile/containerfile
- → then, we do a “play kube” from the ide, just with podman
- → we show it works
- → we show we can deploy it onto kubernetes with a kind cluster.

**Comment by cdrage** [comment]
Internal Google Doc has been created with @slemeur and we've outlined what needs to be completed for a demo scenario / what examples to use.

**Comment by cdrage** [comment]
Did some research on the example finally for https://github.com/redis-developer/beats-by-redis and we unfortunatley won't be able to use it.

Within the code it uses websockets directly and unless you are using the Redis cloud interface, you get this error when deploying locally, which.. won't work unless you use a secure HTTPS / TLS connection which wouldn't be possible locally:

```
purchases.js:170 WebSocket connection to 'ws://localhost/websocket' failed: 
(anonymous) @ purchases.js:170
```

It's the reason why the example had the "local" redis commented out. 

Other observations: The application also has a sign up / login screen which would require some explanation / extra steps (click on sign up, enter a random username and password) in order to use the beats-by-redis.

I did a lot of testing and ended up finding this "guestbook" example which I was able to modify / use the binary to showcase a Redis example with a master / replica as well as a front end: https://github.com/redhat-developer/podman-desktop-demo/pull/34

This includes:
- master and replica redis cluster that syncs with each other
- be able to access the terminal and make commands directly with redis
- real time updates to the GUI
- shows env and file information (would be useful to see if we are on docker / podman / kubernetes!)

I did however come across two issues which are unfortunately blocking the above steps for "Play Kube" and "Deploy to Kube" to a Kind cluster:

https://github.com/containers/podman-desktop/issues/4724
https://github.com/containers/podman-desktop/issues/4723

Which blocks the last two steps. I'm able to write a blog post / tutorial on how to use Compose, but we'd be unable to showcase the Kubernetes portion until the two above issues are fixed.

**Comment by cdrage** [comment]
I ran into another issue too with converting to Kubernetes, to be honest the conversion from the group of compose containers to Kubernetes is done.. poorly with regards to services and being able to access them through the namespace.

For example, if we were to deploy only ONE container, it would be converted correctly. But say if we were to deploy multiple containers where they would be able to access each-other via services? For example, within a `frontend` container, you can ping `redis-master`, it will not work due to the way we are converting services.

When converting this example:


```yaml
services:

  redis-master:
    container_name: redis-master
    image: redis:latest
    ports:
      - "6379"

  redis-replica:
    container_name: redis-replica
    image: redis:latest
    ports:
      - "6379"
    command: redis-server --replicaof redis-master 6379
    depends_on:
      - redis-master

  frontend:
    container_name: frontend
    image: cdrage/frontend
    #build: ./frontend
    ports:
      - "8080:8080"
```

I'm unable to access the "redis-replica" service as well as the "redis-master" service because Podman Desktop / Podman generates the following service:

```
apiVersion: v1
kind: Service
metadata:
  name: redis-replica-pod-35967
  namespace: default
spec:
  ports:
    - name: redis-replica-pod-35967
      protocol: TCP
      port: 35967
      targetPort: 6379
  selector:
    app: redis-replica-pod
```

When it should be creating this (from the kompose conversion):

```
apiVersion: v1
kind: Service
metadata:
  name: redis-replica
spec:
  ports:
    - name: "6379"
      port: 6379
      targetPort: 6379
```

Improvements is:
- Port should be both (not just TCP)
- Port should be the same as target port so that it is accessible by the kubernetes pod. For example, within the compose yaml the frontend service wants to access redis-replica:6379. If deploying via podman containers it will be accessible. It should also still be able to access redis-replica:6379 when  it's in a pod. To do so a service needs to be made with the same name as well as port number. Right now, it is not.
- Name is random? It should be `redis-replica` not `redis-replica-pod-35967`

TLDR: When working with multiple containers with compose, it does not work well at all when converting to Kubernetes. 1 container is okay though.

**Comment by cdrage** [comment]
For Kubernetes we unfortunately have three blockers if we want multi-container compose applications to work correctly:
- [ ] Container names are not converting correctly https://github.com/containers/podman-desktop/issues/4724
- [ ] UI issue showing up https://github.com/containers/podman-desktop/issues/4723
- [ ] Bad service conversion doesn't allow containers to communicate with each other: https://github.com/containers/podman-desktop/issues/4725

**Comment by cdrage** [comment]
I'm going to continue to go ahead and integrate an example (guestbook + redis): https://github.com/redhat-developer/podman-desktop-demo/pull/34 and publish a draft of the blog with steps up to the Kubernetes portion.
